### DISCUSSION TITLE
Is it possible to make custom folder structure in Nuxt?

### DISCUSSION THREAD

**Discussion by Rusinas** [nuxt/nuxt#16626] [Questions]
Probably the pretty frequently asked question, but I didn't manage to find any information about it. I want to use my scrutcure for project, because the Nuxt's structure is unusubale for big projects and becomes a big mess pretty quick. 

The only thing that need to be changed in such scenario is the router - just create routes manually as in Vue. Is it possible?

**Comment by TheAlexLichter** [comment]
You can rename all folders and even use your own router via https://github.com/nuxt/framework/pull/7129

**Reply by Rusinas** [reply]
It's a great idea to use my own router, but it's not clear how exactly I can do it from that PR :(

**Reply by TheAlexLichter** [reply]
1. Create a file at `app/router.options.ts`
2. Define your own routes, e.g. via:

```ts
export default <RouterOptions> {
  routes: (_routes) => [
    {
      name: 'home',
      route: '/',
      component: () => import('~/pages/home.vue')
    }
  ],
}
```

Be aware that you can not "fully" replace the router though, but routes, history and more.

**Reply by Rusinas** [reply]
Yeah, I figured that out myself, even though RouterOptions doesn't seem to accept the "routes" field. And yeah, this is not working :( Got this:
`[Vue warn]: Failed to resolve component: router-view
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <App key=1 > 
  at <NuxtRoot>`

But it seems it doesn't even catch my configuration.

**Reply by TheAlexLichter** [reply]
Make sure you are on the latest RC and create the file in the `app` subdirectory. It works fine in several of my projects ☺️

**Reply by Rusinas** [reply]
I genuinely don't have a clue what I am doing wrong. Sorry for bothering you, but could you check please?
https://github.com/Rusinas/nuxt3-custom-router

**Reply by Rusinas** [reply]
@manniL or maybe you can just share you base config?

**Reply by TheAlexLichter** [reply]
My bad! Right now it seems you need the `pages` folder with dummy content - https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions8098

Opened a bug report in https://github.com/nuxt/framework/issues/8177

**Reply by Rusinas** [reply]
Oh, got it! Thank you so much! This solution is really weird and unobvious as hell, hope it fill be fixed in upcoming releases :) Thank you once again! Didn't test it yet but I think that is it
